services/payments/secrets.go

```diff
@@ -39,6 +44,44 @@ type ChainAddress struct {
 	Approvals []string `ion:"approvals"`
 }
 
+// Vault represents a key which has been broken into shamir shares and is used for encrypting
+// secrets.
+type Vault struct {
+	PublicKey      string   `ion:"publicKey"`
+	Creator        string   `ion:"creator"`
+	Approvals      []string `ion:"approvals"`
+	Threshold      int      `ion:"threshold"`
+	OperatorKeys   []string `ion:"operatorKeys"`
+	IdempotencyKey string   `ion:"idempotencyKey"`
+	// must be unexported. these values should never be presisted to QLDB
+	shares paymentLib.CreateVaultResponse
+}
+
+// SetIdempotencyKey takes and existing vault, checks if it has valid threshold and operators, and
+// generates a UUIDv5 using our Payment Service namespace, threshold, and operators.
```

Comment:
Won't this make it not possible to generate a new vault in DR scenario without modifying the operators or threshold?

Reply:
Yes I hit this during testing and added the public key to the idempotency generation.